R/pkg/R/schema.R

```diff
@@ -157,6 +157,30 @@ checkType <- function(type) {
                 checkType(valueType)
                 return()
               }
+            },
+            s = {
+              # Struct type
+              m <- regexec("^struct<(.+)>$", type)
```

Comment:
We already use JSON to serialize the schema, why not use it here? (It was, but deleted later, you could find it in the history)